tests.cpp

```diff
@@ -0,0 +1,40 @@
+#include <stdio.h>
+
+#include <iostream>
+using namespace std;
+
+#include "mcp23017.h"
+#include "serial.h"
+
+// Test functions go here
+int mcp23017_test() {
+  // initialize the device
+  Mcp23017 dev = Mcp23017(2, 0x24);
+  // begin the device dev.begin(array of 16 bits)
+  if (dev.begin(random_array)) return 1;
+  // write one output to bank a, and one to bank b
+  dev.set_state(3, 1);

```

Comment:
lets also write these pins to 0's as well just to get some extra assurance

Let's also refactor these so we check the pins immediately after we set them instead of doing all of the sets and then all of the gets